server/app/mutations/grid_certificates/get_certificate.rb

```diff
@@ -54,10 +57,8 @@ def execute
           end
         end
 
-        Timeout::timeout(30) {
-          info 'waiting for DNS validation...'
-          sleep 1 until challenge.verify_status == 'valid'
-          info 'DNS validation complete'
+        wait_until!("domain verification for #{domain} is valid", interval: 1, timeout: 30, threshold: 10) {
+          challenge.verify_status == 'valid'
```

Comment:
From the [README](https://github.com/unixcharles/acme-client):
```
# If authorization.verify_status is 'invalid', you can get at the error
# message only through the failed challenge.
authorization.verify_status # => 'invalid'
authorization.http01.error # => {"type" => "...", "detail" => "..."}
```

It would be nice to see those errors? It doesn't seem like they would get raised as `Acme::Client:Error`s?